hcl2template/types.variables.go

```diff
@@ -200,6 +200,12 @@ func (variables *Variables) decodeVariableBlock(block *hcl.Block, ectx *hcl.Eval
 		})
 	}
 
+	// It's possible no type attribute was assigned so lets make
+	// sure we have a valid type otherwise there will be issues parsing the value.
+	if res.Type == cty.NilType {
+		res.Type = res.DefaultValue.Type()
+	}
+
```

Comment:
Nice one ! I thought I had done this before; this is good !

Do you think it would be a better idea to group this into the `if def, ok := attrs["default"]; ok {` block ?

Reply:
Ah yes, I like that suggestion. So would this go under line 184 `res.DefaultValue = defaultValue`? 

Essentially, if we've gotten this far and still have no type take the type of whatever value we have just set to res.DefaultValue, correct? No need to try and test the type conversion since we are pulling it directly from what the code has determined the value to be.

Reply:
Yes I think right after `res.DefaultValue = defaultValue` would be a nice spot 🙂